src/server-core/src/lib.rs

```diff
@@ -69,6 +70,52 @@ impl Connection {
     pub fn uuid_handle(&self) -> ConnectionUuidHandle {
         ConnectionUuidHandle(Arc::clone(&self.conn_uuid))
     }
+
+    /// Attempts to parse a proxy header from the tcp_stream.
+    /// If none is found or it is unable to be parsed None will
+    /// be returned. If a header is found it will be returned and its
+    /// bytes will be removed from the stream.
+    ///
+    /// It is possible an invalid header was sent, if that is the case
+    /// any downstream service will be responsible for returning errors
+    /// to the client.
+    pub async fn take_proxy_header(&mut self) -> Option<ProxyHeader> {
```

Comment:
I think you probably want to return `Option<ProxiedAddress>` here (https://docs.rs/proxy-header/latest/proxy_header/struct.ProxiedAddress.html). The `ProxyHeader` looks like it only exists to carry around the remainder of the buffer, which isn't actually useful with the nice way that you've got the peeking set up here.